flood_adapt/object_model/tipping_point.py

```diff
@@ -0,0 +1,338 @@
+
+        # crete scenarios for each SLR value
+        scenarios = {
+            f"slr_{slr}": {
+                "name": f"slr_{slr}",
+                "event": self.attrs.event_set,
+                "projection": f"{self.attrs.projection}_slr{slr}",
+                "strategy": self.attrs.strategy,
+            }
+            for slr in self.attrs.sealevelrise
+        }
+
+        # create subdirectories for each scenario and .toml files
+        for scenario in scenarios.keys():
+
+            scenario_obj = Scenario.load_dict(
+                scenarios[scenario], Database().input_path
+            )
+            scen_exists = self.check_scenarios_exist(scenario_obj)
+
+            if scen_exists:
+                # make a dict with name and object
+                self.scenarios[scen_exists[0]] = {
+                    "name": scen_exists[0],
+                    "object": scenario_obj,
+                }
+            else:
+                Database().scenarios.save(scenario_obj)
+                self.scenarios[scenario_obj.attrs.name] = {
+                    "name": scenario_obj.attrs.name,
+                    "object": scenario_obj,
+                }
+
+        self.attrs.scenarios = list(self.scenarios.keys())
+        self.save(
+            filepath=Database().input_path
+            / "tipping_points"
+            / self.attrs.name
+            / f"{self.attrs.name}.toml"
+        )  # for later when we have a database_tp: TODO: Database().tipping_points.save(self)
+
+    def run_tp_scenarios(self):
+        """Run all scenarios to determine tipping points."""
+        for name, scenario in self.scenarios.items():
+            scenario_obj = scenario["object"]
+
+            if self.attrs.status == TippingPointStatus.reached:
+                self.scenarios[name]["tipping point reached"] = True
+                continue
+
+            if not self.scenario_has_run(scenario_obj):
+                scenario_obj.run()
+
+            # Check the tipping point status
+            if self.check_tipping_point(scenario_obj):
+                self.attrs.status = TippingPointStatus.reached
+                self.scenarios[name]["tipping point reached"] = True
+            else:
+                self.scenarios[name]["tipping point reached"] = False
+
+        tp_path = self.results_path.joinpath(self.attrs.name)
+
+        # Save results - make directory if it doesn't exist
+        if not tp_path.is_dir():
+            tp_path.mkdir(parents=True)
+
+        tp_results = pd.DataFrame.from_dict(self.scenarios, orient="index").reset_index(
+            drop=True
+        )
+        tp_results["sea level"] = [
+            float(i) / 10 for i in self.attrs.sealevelrise
+        ]  # TODO: hardcoded for SLR in Meters, fix later
+        # calculate the sea level at the tipping point threshold
+        # TODO: not super clear but maybe talk to @luuk about how to better show the table
```

Comment:
@LuukBlom, the way we thought to have the csv file is not really aligned to how they use the tipping points in the pathway generator. I'm not sure if they are using in the best way anyway, so I added some information but in a different format. Would you have any thoughts?

Reply:
Also something important to consider, but I still have to think better on what is the best solution: they use different strategies in the same place (as seen in the table), whereas we have a single strategy for all the scenarios. Should we adapt the tipping point object or make a wrapper to combine all these in plots and tables?